tools/distpartitioning/convert_partition.py

```diff
@@ -450,56 +459,156 @@ def create_dgl_object(
+        part_global_ids = part_global_ids[reshuffle_nodes]
+        ntype, per_type_ids = id_map(part_global_ids)
+
+        # continue with the graph creation
+        node_attr = {}
+        node_attr[dgl.NTYPE] = th.as_tensor(
+            ntype, dtype=RESERVED_FIELD_DTYPE[dgl.NTYPE]
+        )
+        node_attr[dgl.NID] = th.as_tensor(uniq_ids[reshuffle_nodes])
+        node_attr["inner_node"] = th.as_tensor(
+            inner_nodes[reshuffle_nodes],
+            dtype=RESERVED_FIELD_DTYPE["inner_node"],
+        )
 
-    # continue with the graph creation
-    part_graph.ndata[dgl.NTYPE] = th.as_tensor(
-        ntype, dtype=RESERVED_FIELD_DTYPE[dgl.NTYPE]
-    )
-    part_graph.ndata[dgl.NID] = th.as_tensor(uniq_ids[reshuffle_nodes])
-    part_graph.ndata["inner_node"] = th.as_tensor(
-        inner_nodes[reshuffle_nodes], dtype=RESERVED_FIELD_DTYPE["inner_node"]
-    )
+        is_homo = is_homogeneous(ntypes, etypes)
+        orig_nids = None
+        orig_eids = None
+        if return_orig_nids:
+            orig_nids = {}
+            for ntype, ntype_id in ntypes_map.items():
+                mask = th.logical_and(
+                    node_attr[dgl.NTYPE] == ntype_id,
+                    node_attr["inner_node"],
+                )
+                orig_nids[ntype] = th.as_tensor(per_type_ids[mask])
+        if return_orig_eids:
+            orig_eids = {}
+            for etype, etype_id in etypes_map.items():
+                mask = th.logical_and(
+                    type_per_edge == etype_id,
+                    edge_attr["inner_edge"],
+                )
+                orig_eids[_etype_tuple_to_str(etype)] = th.as_tensor(
+                    global_edge_id[mask]
+                )
+        edge_type_to_id = (
+            None
+            if is_homo
+            else {
+                gb.etype_tuple_to_str(etype): etid
+                for etype, etid in etypes_map.items()
+            }
+        )
 
-    orig_nids = None
-    orig_eids = None
-    if return_orig_nids:
-        orig_nids = {}
-        for ntype, ntype_id in ntypes_map.items():
-            mask = th.logical_and(
-                part_graph.ndata[dgl.NTYPE] == ntype_id,
-                part_graph.ndata["inner_node"],
-            )
-            orig_nids[ntype] = th.as_tensor(per_type_ids[mask])
-    if return_orig_eids:
-        orig_eids = {}
-        for etype, etype_id in etypes_map.items():
-            mask = th.logical_and(
-                part_graph.edata[dgl.ETYPE] == etype_id,
-                part_graph.edata["inner_edge"],
-            )
-            orig_eids[_etype_tuple_to_str(etype)] = th.as_tensor(
-                global_edge_id[mask]
-            )
+        part_local_src_id, part_local_dst_id = th.tensor(
+            part_local_src_id, dtype=th.int64
+        ), th.tensor(part_local_dst_id, dtype=th.int64)
+        size = max(part_local_src_id.max(), part_local_dst_id.max()) + 1
+        adj_matrix = dglsp.from_coo(
```

Comment:
refer to https://github.com/dmlc/dgl/blob/b3eacd22d7c8e772d29d42011d9a569b64a89579/tests/python/pytorch/graphbolt/impl/test_fused_csc_sampling_graph.py#L1439-L1442